azurerm/internal/services/compute/client.go

```diff
@@ -0,0 +1,67 @@
+package compute
+
+import (
+	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2018-06-01/compute"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/common"
+)
+
+type Client struct {
+	AvailabilitySetsClient     compute.AvailabilitySetsClient
+	DisksClient                compute.DisksClient
+	GalleriesClient            compute.GalleriesClient
+	GalleryImagesClient        compute.GalleryImagesClient
+	GalleryImageVersionsClient compute.GalleryImageVersionsClient
+	ImagesClient               compute.ImagesClient
+	SnapshotsClient            compute.SnapshotsClient
+	UsageClient                compute.UsageClient
+	VMExtensionImageClient     compute.VirtualMachineExtensionImagesClient
+	VMExtensionClient          compute.VirtualMachineExtensionsClient
+	VMScaleSetClient           compute.VirtualMachineScaleSetsClient
+	VMClient                   compute.VirtualMachinesClient
+	VMImageClient              compute.VirtualMachineImagesClient
```

Comment:
in a subsequent PR (since this affects everything) in retrospect all of the clients here should be pointers; so that we're forced to initialize these (which ensures the authorizer is added, which in turn helps provider developers debug issues)